lnx-engine/search-index/src/schema.rs

```diff
@@ -0,0 +1,538 @@
+                rejected_fields.len(),
+                rejected_fields.join(", "),
+            ));
+            }
+        }
+
+        {
+            let mut rejected_fields = vec![];
+            for field_name in self.search_fields.iter() {
+                if !self.has_field(field_name) {
+                    rejected_fields.push(field_name.to_string());
+                }
+            }
+
+            if !rejected_fields.is_empty() {
+                return Err(anyhow!(
+                "key 'search_fields' contain {} fields that are not defined in the schema: {}",
+                rejected_fields.len(),
+                rejected_fields.join(", "),
+            ));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Calculated for SchemaContext {
+    fn calculate_once(&mut self) -> Result<()> {
+        self.required_fields = HashSet::from_iter(
+            self.fields
+                .iter()
+                .filter_map(
+                    |(name, info)| if info.is_required() { Some(name) } else { None },
+                )
+                .cloned(),
+        );
+
+        self.multi_value_fields = HashSet::from_iter(
+            self.fields
+                .iter()
+                .filter_map(
+                    |(name, info)| if info.is_multi() { Some(name) } else { None },
+                )
+                .cloned(),
+        );
+
+        Ok(())
+    }
+}
+
+impl SchemaContext {
+    pub fn has_field(&self, field_name: &str) -> bool {
+        self.fields.contains_key(field_name)
+    }
+
+    #[inline]
+    pub fn boost_fields(&self) -> &HashMap<String, Score> {
+        &self.boost_fields
+    }
+
+    #[inline]
+    pub fn fields(&self) -> &HashMap<String, FieldDeclaration> {
+        &self.fields
+    }
+
+    #[inline]
+    pub fn required_fields(&self) -> &HashSet<String> {
+        &self.required_fields
+    }
+
+    #[inline]
+    pub fn multi_value_fields(&self) -> &HashSet<String> {
+        &self.multi_value_fields
+    }
+
+    /// Checks and asserts that the fields defined by Tantivy are also the same set of fields
+    /// defined in the schema.
+    ///
+    /// These should never be off unless someone has manually modified the data.
+    pub fn assert_existing_schema_matches(&self, existing: &Schema) -> Result<()> {
+        let defined = self.as_tantivy_schema();
+        let defined_fields: Vec<&str> = defined
+            .fields()
+            .map(|(f, _)| defined.get_field_name(f))
+            .collect();
+
+        let existing_fields: Vec<&str> = existing
+            .fields()
+            .map(|(f, _)| existing.get_field_name(f))
+            .collect();
+
+        let defined_fields_set: HashSet<&str> =
+            HashSet::from_iter(defined_fields.into_iter());
+        let existing_fields_set: HashSet<&str> =
+            HashSet::from_iter(existing_fields.into_iter());
+
+        let union: Vec<&str> = defined_fields_set
```

Comment:
Shouldn't this variable be called `diff` or something?

Reply:
it does indeed, left over from testing 😓 Has been changed in my local copy.